pkg/render/node.go

```diff
@@ -945,16 +954,26 @@ func (c *nodeComponent) nodeMetricsService() *v1.Service {
 		Spec: v1.ServiceSpec{
 			Selector: map[string]string{"k8s-app": "calico-node"},
 			Type:     v1.ServiceTypeClusterIP,
-			Ports: []v1.ServicePort{
-				v1.ServicePort{
-					Name:       "calico-metrics-port",
-					Port:       nodeMetricsPort,
-					TargetPort: intstr.FromInt(int(nodeMetricsPort)),
-					Protocol:   v1.ProtocolTCP,
-				},
-			},
 		},
 	}
+
+	// If the caller hasn't specified a NodeMetricsPort, default it to 9081.
+	if c.cr.Spec.NodeMetricsPort != nil {
+		if *c.cr.Spec.NodeMetricsPort == 0 {
```

Comment:
I don't think this check belongs here in the render code. Mostly because the render code should be doing exactly what the user has specified in the CR (i.e., if the user specified `0`, then we should use port `0`).

Reply:
This was needed for the UTs, they call into `Render()` directly.

Reply:
I think we should adjust the UTs then, since they appear to not be a realistic test :)

Reply:
Ok